validator/src/commands/run/args/pub_sub_config.rs

```diff
@@ -11,20 +11,77 @@ use {
+static DEFAULT_TEST_RPC_PUBSUB_MAX_ACTIVE_SUBSCRIPTIONS: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .max_active_subscriptions
+        .to_string()
+});
 
 static DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS: LazyLock<String> =
     LazyLock::new(|| PubSubConfig::default().queue_capacity_items.to_string());
+static DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .queue_capacity_items
+        .to_string()
+});
 
 static DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_BYTES: LazyLock<String> =
     LazyLock::new(|| PubSubConfig::default().queue_capacity_bytes.to_string());
+static DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_BYTES: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .queue_capacity_bytes
+        .to_string()
+});
 
 const DEFAULT_RPC_PUBSUB_WORKER_THREADS: &str = "4";
+static DEFAULT_TEST_RPC_PUBSUB_WORKER_THREADS: LazyLock<String> = LazyLock::new(|| {
+    PubSubConfig::default_for_test_validator()
+        .worker_threads
+        .to_string()
+});
 
 #[cfg_attr(test, qualifiers(pub(crate)))]
 static DEFAULT_RPC_PUBSUB_NUM_NOTIFICATION_THREADS: LazyLock<String> =
     LazyLock::new(|| get_thread_count().to_string());
 
-pub(crate) fn args<'a, 'b>() -> Vec<Arg<'a, 'b>> {
+pub(crate) fn args<'a, 'b>(test_validator: bool) -> Vec<Arg<'a, 'b>> {
+    let rpc_pubsub_notification_threads = Arg::with_name("rpc_pubsub_notification_threads")
+        .long("rpc-pubsub-notification-threads")
+        .takes_value(true)
+        .value_name("NUM_THREADS")
+        .validator(is_parsable::<usize>)
+        .help(
+            "The maximum number of threads that RPC PubSub will use for generating notifications. \
+             0 will disable RPC PubSub notifications",
+        );
+    let (rpc_pubsub_notification_threads, strings) = if test_validator {
+        (
+            rpc_pubsub_notification_threads.default_value(&DEFAULT_TEST_RPC_PUBSUB_WORKER_THREADS),
+            [
+                &DEFAULT_TEST_RPC_PUBSUB_MAX_ACTIVE_SUBSCRIPTIONS,
+                &DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS,
+                &DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_BYTES,
+            ],
+        )
+    } else {
+        (
+            rpc_pubsub_notification_threads
+                .default_value_if(
+                    "full_rpc_api",
+                    None,
+                    &DEFAULT_RPC_PUBSUB_NUM_NOTIFICATION_THREADS,
+                )
+                .requires("full_rpc_api"),
+            [
+                &DEFAULT_RPC_PUBSUB_MAX_ACTIVE_SUBSCRIPTIONS,
+                &DEFAULT_RPC_PUBSUB_QUEUE_CAPACITY_ITEMS,
+                &DEFAULT_TEST_RPC_PUBSUB_QUEUE_CAPACITY_BYTES,
+            ],
+        )
+    };
+    let [default_rpc_pubsub_max_active_subscriptions, default_rpc_pubsub_queue_capacity_items, default_rpc_pubsub_queue_capacity_bytes] =
+        strings;
```

Comment:
Instead of the intermediate `strings`, can you just do the assignment directly (and obviously run `cargo fmt`) ?
```rust
let (rpc_pubsub_notification_threads, default_rpc_pubsub_max_active_subscriptions, default_rpc_pubsub_queue_capacity_items, default_rpc_pubsub_queue_capacity_bytes) = if test_validator { ...
```